firebase-common/src/main/java/com/google/firebase/concurrent/FirebaseExecutors.java

```diff
@@ -33,6 +35,91 @@ public static Executor newSequentialExecutor(Executor delegate) {
     return new SequentialExecutor(delegate);
   }
 
+  /**
+   * Returns a {@link Executor} that limits the number of running tasks at a given time.
+   *
+   * <p>The executor uses delegate in order to {@link Executor#execute(Runnable) execute} each task,
+   * and does not create any threads of its own.
+   *
+   * @param delegate {@link Executor} used to execute tasks
+   * @param concurrency max number of tasks executing concurrently
+   */
+  public static Executor newLimitedConcurrencyExecutor(Executor delegate, int concurrency) {
```

Comment:
Note that currently it won't be possible to change the concurrency dynamically, might be worth returning a `LimitedConcurrencyExecutor` instead?

Reply:
IMHO It's OK to leave it this way.